toqito/channel_metrics/fidelity_of_separability.py

```diff
@@ -0,0 +1,198 @@
+        \left\{\begin{array}
+                [c]{c}%
+                \operatorname{Tr}[\Pi_{A^{\prime}A}^{\operatorname{sym}} \operatorname{Tr}_{R}[T_R(\psi_{RAB})\Gamma^{\mathcal{E}^{k}}_{RA^{\prime}_1}]]:\\%
+                \operatorname{Tr}_{A^{\prime k}}[\Gamma^{\mathcal{E}^{k}}_{RA^{\prime k}}]=I_R,\\
+                \Gamma^{\mathcal{E}^{k}}_{RA^{\prime k}}=\mathcal{P}_{A^{\prime k}}(\Gamma^{\mathcal{E}^{k}}_{RA^{\prime k}}),\\ 
+                T_{A^{\prime}_{1\cdots j}}(\Gamma^{\mathcal{E}^{k}_{RA^{\prime k}}}) \geq 0 \quad \forall j\leq k
+        \end{array}\right\}
+        \end{multline}
+
+    :math:`\frac{1}{2}(1+\widetilde{F}_s^2(\rho_{AB}))` is the quantity to be
+    approximated but this function returns
+    :math:`\widetilde{F}_s^2(\rho_{AB})`.
+
+    :math:`\operatorname{Tr}[\Pi_{A^{\prime}A}^{\operatorname{sym}} \operatorname{Tr}_{R}[T_R(\psi_{RAB})\Gamma^{\mathcal{E}^{k}}_{RA^{\prime}_1}]]`
+    is the maximization problem subject to PPT & k-extendibile channel
+    constraints.
+
+    :math:`\Gamma^{\mathcal{E}^{k}}_{RA^{\prime}}` is Choi operator of
+    entanglement breaking channel :math:`\mathcal{E}^{k}`.
+
+    :math:`\mathcal{P}_{A^{\prime k}}` is the permutation operator over
+    k-extensions :math:`A^{\prime k}`.
+
+    The other constraints are due to the PPT condition [Per96]_.
+
+    Examples
+    ==========
+    Let's consider a density matrix of a state that we know is pure &
+    separable. :math:`|000 \rangle = |0 \rangle \otimes |0 \rangle \otimes |0 \rangle`.
+    
+    The expected approximation of fidelity of separability is the maximum
+    value possible i.e. very close to 1.
+
+    .. math::
+        \rho_{AB} = |000 \rangle \langle 000|
+
+    .. code-block:: python
+
+        from toqito.channel_metrics import fidelity_of_separability
+        from toqito.matrix_ops import tensor
+        from toqito.states import basis
+
+        state = tensor(basis(2, 0), basis(2, 0), basis(2, 0))
+        rho = state*state.conj().T
+        expected_value = fidelity_of_separability(rho, [2,2,2])
+
+    >>> expected_value = 0.9999999979949119
+
+
+    References
+    ==========
+    .. [Hay12] Hayden, Patrick et.al.
+        "Two-message quantum interactive proofs and the quantum separability problem."
+        Proceedings of the 28th IEEE Conference on Computational Complexity, pages 156-167.
+        https://arxiv.org/abs/1211.6120
+
+    .. [Per96] Peres, Asher.
+        "Separability Criterion for Density Matrices"
+        https://arxiv.org/abs/quant-ph/9604005
+
+    .. [Phil23] Philip, Aby et.al.
+        "Quantum Steering Algorithm for Estimating Fidelity of Separability"
+        https://arxiv.org/abs/2303.07911
+
+    .. [TBWat18] Watrous, John.
+        “The Theory of Quantum Information”
+        Cambridge University Press, 2018
+
+    Args:
+        psi: the density matrix for the tripartite state of interest psi_{BAR}
+        psi_dims: the dimensions of System A, B & R in
```

Comment:
Nitpick: comma after "B".

Note: I really should hook up automated linting to save you from having to make all these minor changes :). I'll make myself a note to do that to spare future developers the annoyance of me making these types of comments.

Reply:
I tried to use black & flake8 for the entire `toqito` repo. There were quite a few reported issues. 

Plus, I was not sure what things you wanted to be highlighted as an error & what was acceptable. For example, there are quite a few lines in this PR itself that are longer than what is allowed by flake8.

Reply:
Yeah, `black` out of the box might make certain changes that we don't necessarily want. It's on my list of things to do to have a configuration file that is a bit more sensible to conform to the `toqito` project. Running `black` on the project is an eventual goal, but probably outside of the scope for this issue :)

Reply:
> It's on my list of things to do to have a configuration file that is a bit more sensible to conform to the toqito project.

Ok. If you want, I can also work on getting to this stage. The first step would be manually making the changes that flake8 complains about. 

I am used to how `mitiq` has [different workflows](https://github.com/unitaryfund/mitiq/tree/master/.github/workflows) to check code style, running tests for the code (which toqito does have now with a recent PR) & another to build the documentation from a PR.

Reply:
Yes, `mitiq` has quite a bit more checks in place which is great. Any inspiration you'd like to take from the `mitiq` project that could benefit the `toqito` workflow I'm all for!